packages/ui/src/domainModules/localityService/selectors.js

```diff
@@ -29,3 +35,44 @@ export const getHasCuratedLocalities = createSelector(
     return Object.keys(curatedLocalities).length > 0
   }
 )
+
+const createDropdownSelector = groupFilter => {
+  return createSelector(
+    [getCuratedLocalities, getSecondArgument],
+    (curatedLocalities, searchQuery = '') => {
+      const lowerCaseSearchQuery = searchQuery.toLowerCase()
+      const firstLetterMatches = Object.values(curatedLocalities)
+        .filter(({ group }) => group === groupFilter)
+        .map(({ id, name }) => {
```

Comment:
Would make sense to save the result of this map in a variable and then apply the two filters (match beginning or any part) on that, so we don't do the same Object.values-filter-map twice